File: PulseSocial-Frontend/src/services/followService.js
import api from "../api/axios";

export const followUser = (followingId, followerId) => {
    const url = followerId ? `/follows/${followingId}?followerId=${followerId}` : `/follows/${followingId}`;
    return api.post(url);
};

export const getFollowers = (userId) => {
    return api.get(`/follows/followers/${userId}`);
};

export const getFollowing = (userId) => {
    return api.get(`/follows/following/${userId}`);
};

export const getFollowersCount = (userId) => {
    return api.get(`/follows/followers/count/${userId}`);
};

export const getFollowingCount = (userId) => {
    return api.get(`/follows/following/count/${userId}`);
};

File: src/main/java/com/pulsesocial/service/FollowService.java
package com.pulsesocial.service;

import java.util.List;
import com.pulsesocial.dto.UserResponse;

public interface FollowService {
    String followUser(Long followerId, Long followingId);
    List<UserResponse> getFollowers(Long userId);
    List<UserResponse> getFollowing(Long userId);
    long getFollowersCount(Long userId);
    long getFollowingCount(Long userId);
}
